internal/repository/ent/subscription.go

```diff
@@ -922,6 +923,74 @@ func (r *subscriptionRepository) ListByIDs(ctx context.Context, subscriptionIDs
 	return result, nil
 }
 
+func (r *subscriptionRepository) FindEligibleSubscriptionsForGrant(ctx context.Context, grant *creditgrant.CreditGrant) ([]*domainSub.Subscription, error) {
```

Comment:
No need directly fetch list or one by one each subscription as a single credit grant is related to a single subscription only